python/pyarrow/tests/test_compute.py

```diff
@@ -2481,6 +2493,10 @@ def _check_temporal_rounding(ts, values, unit):
 
 @pytest.mark.skipif(sys.platform == "win32" and not util.windows_has_tzdata(),
                     reason="Timezone database is not installed on Windows")
+@pytest.mark.xfail(
+    sys.platform == "emscripten",
+    reason="Emscripten datetime is implemented in Javascript and works differently"
+)
```

Comment:
Out of curiosity, do you remember what exactly was failing here? Because this test is rounding timestamps, which I would think is mostly implemented by our own code and not relying on something platform specific

Reply:
Ah, I've got it, it's something to do with timezones and daylight saving, I think it isn't actually in the emscripten javascript that the bug is happening. I think there's an error with DST.

I can't work out where the bug is though, whether it is a bug in arrow that only happens in emscripten, or an emscripten libc++ bug or what.

If I run the below code, which takes a timestamp near the daylight savings time change, then pandas and arrow disagree as to whether some dates should be dst or not.

```
import pyarrow as pa
import pyarrow.compute as pc
import pandas as pd
for x in range(0,20):
	t=pd.to_datetime("%d-10-26 09:39:00.316686848+01:00 "%(x+1990)).tz_convert("Europe/London")
	s=pa.scalar(t)	
	print(t,s,t.dst(),pc.is_dst(s))
```
Output looks like this


|Pandas_timestamp|Arrow_timestamp|pandasDST_offset|pyarrow_dst_offset|
|-----------|------------|--------------|-----------|
|1990-10-26 09:39:00.316686848+01:00|1990-10-26 09:39:00.316686+01:00|1:00:00|True|
|1991-10-26 09:39:00.316686848+01:00|1991-10-26 09:39:00.316686+01:00|1:00:00|True|
|1992-10-26 08:39:00.316686848+00:00|1992-10-26 08:39:00.316686+00:00|0:00:00|False|
|1993-10-26 08:39:00.316686848+00:00|1993-10-26 08:39:00.316686+00:00|0:00:00|False|
|1994-10-26 08:39:00.316686848+00:00|1994-10-26 08:39:00.316686+00:00|0:00:00|False|
|1995-10-26 08:39:00.316686848+00:00|1995-10-26 08:39:00.316686+00:00|0:00:00|False|
|1996-10-26 09:39:00.316686848+01:00|1996-10-26 09:39:00.316686+01:00|1:00:00|False|
|1997-10-26 08:39:00.316686848+00:00|1997-10-26 08:39:00.316686+00:00|0:00:00|False|
|1998-10-26 08:39:00.316686848+00:00|1998-10-26 08:39:00.316686+00:00|0:00:00|False|
|1999-10-26 09:39:00.316686848+01:00|1999-10-26 09:39:00.316686+01:00|1:00:00|False|
|2000-10-26 09:39:00.316686848+01:00|2000-10-26 09:39:00.316686+01:00|1:00:00|False|
|2001-10-26 09:39:00.316686848+01:00|2001-10-26 09:39:00.316686+01:00|1:00:00|False|
|2002-10-26 09:39:00.316686848+01:00|2002-10-26 09:39:00.316686+01:00|1:00:00|False|
|2003-10-26 08:39:00.316686848+00:00|2003-10-26 08:39:00.316686+00:00|0:00:00|False|
|2004-10-26 09:39:00.316686848+01:00|2004-10-26 09:39:00.316686+01:00|1:00:00|False|
|2005-10-26 09:39:00.316686848+01:00|2005-10-26 09:39:00.316686+01:00|1:00:00|False|
|2006-10-26 09:39:00.316686848+01:00|2006-10-26 09:39:00.316686+01:00|1:00:00|False|
|2007-10-26 09:39:00.316686848+01:00|2007-10-26 09:39:00.316686+01:00|1:00:00|False|
|2008-10-26 08:39:00.316686848+00:00|2008-10-26 08:39:00.316686+00:00|0:00:00|False|
|2009-10-26 08:39:00.316686848+00:00|2009-10-26 08:39:00.316686+00:00|0:00:00|False|

Reply:
Aha, nailed it down -

We're hitting this:
https://github.com/HowardHinnant/date/issues/614

because we get our timezone files using the python tzdata package, which builds using the -b slim option, which is not supported yet by HowardHinnant/date

I'm inclined to bundle pre-built timezones into pyarrow build as resources?

Reply:
Ah, good catch. I should have thought about this, because we have discussed related issues in the past. We explicitly don't use the `tzdata` package because of the compatibility issue. Although the discussion at https://github.com/apache/arrow/issues/31472 / https://github.com/apache/arrow/issues/28868#issuecomment-1378032344 is specifically for Windows (where getting the tzdata through a package would be especially useful, given we now require users to download the text database manually), and from re-reading that it seems that the binary format not being supported is only a Windows issue, and for linux it should be fine. But generally for linux the system provides the tz data, so we don't actually test that with the `tzdata` package. I am wondering if we would have similar problems as you encountered here for Linux with `tzdata` .. (given the `slim` / `fat` difference)

> I'm inclined to bundle pre-built timezones into pyarrow build as resources?

I am not entirely sure this would actually be possible.

For Windows, we do provide a helper function to both set the path to the tz database manually to a custom path (`pyarrow.set_timezone_db_path`), and then also a helper to download the data (`pyarrow.util.download_tzdata_on_windows`). 

But that requires you to be able to point to a custom path, and right now that is only possible on Windows AFAIK

Reply:
> because we get our timezone files using the python tzdata package, which builds using the -b slim option, which is not supported yet by HowardHinnant/date

It's still a bit bizarre. 
I checked locally on my Linux in a conda env, and the zoneinfo files provided in `/env/share/zoneinfo` ("system" provided) or `env/lib/python3.11/site-packages/tzdata/zoneinfo` (so from tzdata package) are exactly the same. So if `txdata` provides the slim binaries, I don't understand how this is working ..